test/sequential/test-child-process-execsync.js

```diff
@@ -12,6 +12,13 @@ var start = Date.now();
 var err;
 var caught = false;
 
+// Verify that stderr is not accessed when a bad shell is used
+(function() {
+  assert.throws(function() {
+    execSync('exit -1', {shell: 'no_shell'});
+  }, /spawnSync no_shell ENOENT/);
+})();
+
```

Comment:
If the actual exception is masked, the one who runs the test won't be able to easily identify the problem.
If I move this test at the end, then there is no reason for adding it, since the first test will clearly fail.
The problem with accessing a null stderr is that the test with timeout will:
- fail in ETIMEOUT assert, this way err won't be assigned
- in it's finally clause, the assert for err.status will also throw another exception, since err is null

The test will fail telling that it cannot access the 'status' of an undefined member at "assert(err.status > 128 || err.signal)", which apparently appears to be a problem in the test itself.